iis/tests/test_e2e.py

```diff
@@ -2,15 +2,164 @@
 # All rights reserved
 # Licensed under Simplified BSD License (see LICENSE)
 
+import socket
+import subprocess
+
 import pytest
+import six
 
-from datadog_checks.dev.testing import requires_py3
+from datadog_checks.base import AgentCheck
+from datadog_checks.dev.utils import get_active_env
 from datadog_checks.iis import IIS
+from datadog_checks.iis.service_check import IIS_APPLICATION_POOL_STATE
+
+from .common import CHECK_NAME
+
+
+@pytest.mark.e2e
+def test_e2e(dd_agent_check, aggregator, instance):
+    """
+    Check should run without error if IIS is installed
+    """
+    aggregator = dd_agent_check(instance)
+    if six.PY3:
+        aggregator.assert_service_check('iis.windows.perf.health', IIS.OK)
+
+
+@pytest.fixture
+def iis_host(aggregator, dd_default_hostname):
+    if six.PY2:
+        # the python2 version of the check uses the configured hostname,
+        # which is the hostnane of the docker host.
+        return 'iis_host:{}'.format(socket.gethostname())
+    else:
+        # the python3 version of the check uses the perf counter server field,
+        # which is the hostname of the container.
+        hostname = run_command(['hostname'])
+        return 'iis_host:{}'.format(hostname)
+
+
+def normalize_tags(tags):
+    if six.PY2:
+        # The python2 version of the check calls self.normalize_tag
+        a = AgentCheck()
+        return [a.normalize_tag(tag) for tag in tags]
+    else:
+        return tags
+
+
+def run_command(command):
+    """
+    Run a command in the docker container being used for E2E tests
+    """
+    container_name = 'dd_{}_{}'.format(CHECK_NAME, get_active_env())
+    result = (
+        subprocess.check_output(
+            ['docker', 'exec', container_name] + command,
+        )
+        .decode()
+        .strip()
+    )
+    return result
+
+
+def start_iis():
+    app_pools = ['DefaultAppPool']
+    sites = ['Default Web Site']
+
+    for app_pool in app_pools:
+        run_command(['powershell.exe', '-Command', 'Start-WebAppPool -Name "{}"'.format(app_pool)])
+    for site in sites:
+        run_command(['powershell.exe', '-Command', 'Start-IISSite -Name "{}"'.format(site)])
+
+
+def stop_iis():
+    app_pools = ['DefaultAppPool']
+    sites = ['Default Web Site']
+
+    for app_pool in app_pools:
+        run_command(
+            [
+                'powershell.exe',
+                '-Command',
+                (
+                    'if((Get-WebAppPoolState -Name "{app_pool}").Value -ne "Stopped") {{'
+                    + ' Stop-WebAppPool -Name "{app_pool}"'
+                    + '}}'
+                ).format(app_pool=app_pool),
+            ]
+        )
+    for site in sites:
+        run_command(['powershell.exe', '-Command', 'Stop-IISSite -Name "{}" -Confirm:$false'.format(site)])
```

Comment:
I wonder if something simpler could be done like start and stop whole service (at least for now.

Reply:
tried it, the site stops but the app pool keeps running :(